frontend/src/pages/profilePage.tsx

```diff
@@ -13,19 +14,75 @@ import {
 import { UserContext } from "../contexts/userContext";
 
 import BiFire from "@/assets/bi_fire.png";
-import Frog from "@/assets/frog.png";
+import fire from "@/assets/fire.png";
 import Media from "@/assets/media.png";
+import nature from "@/assets/nature.png";
 import Settings from "@/assets/settings.png";
 import Trophy from "@/assets/trophy.png";
+import water from "@/assets/water.png";
 import ButtonItem from "@/components/ProfileButton";
 import { lightModeColors } from "@/constants/colors";
 
 // import SGDemiBold from "@/assets/fonts/Social-Gothic-DemiBold.otf";
+type Character = {
+  color: string;
+  character: string;
+  characterIcon: ImageSourcePropType;
+};
+
+const characters: Character[] = [
+  {
+    color: "#83B26D",
+    character: "Nature",
+    characterIcon: nature,
+  },
+  {
+    color: "#FFC97E",
+    character: "Fire",
+    characterIcon: fire,
+  },
+  {
+    color: "#FCA397",
+    character: "Water",
+    characterIcon: water,
+  },
+];
 
 export default function ProfilePage() {
   const router = useRouter();
   const { mongoUser } = useContext(UserContext);
+  const selectedCharacter =
+    characters.find((c) => c.character === mongoUser?.character) ?? characters[1];
+  const lessonsCompletedCount = mongoUser?.completedLessons?.length ?? 0;
+
+  const streakDays = useMemo(() => {
+    const completed = mongoUser?.completedLessons ?? [];
```

Comment:
For the streak, can you check journal entries in addition to lessons? So find the streak of contiguous days when the user completed at least one lesson or journal entry